crates/rpc/rpc/src/eth/pubsub.rs

```diff
@@ -372,7 +380,46 @@ where
             futures::stream::iter(headers)
         })
     }
+}
+
+impl<N: NodePrimitives<SignedTx: TxHashRef>, Eth> EthPubSubInner<Eth>
+where
+    Eth: RpcNodeCore<Provider: CanonStateSubscriptions<Primitives = N>>
+        + EthApiTypes<
+            RpcConvert: RpcConvert<Primitives = N, Network = Eth::NetworkTypes, Error = Eth::Error>,
+        >,
+{
```

Comment:
we need to wire this into the match stmt above so we can instead extend the bounds on the existing impl block already, then we dont need to unify them later